honeycombio/honeycomb-opentelemetry-python
# maint: improve ci time

## Which problem is this PR solving?

- slow CI times (because of slow smoke tests)

## Short description of the changes

- switch to a slim docker image instead of alpine
- add metrics export interval of 5s instead of the [default](https://github.com/open-telemetry/opentelemetry-python/blob/3c98eee24aa69b201db542f3472eee28f6fe598c/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py#L447) 60s

There are some posts about alpine being slower than slim for python that lead to me trying it out ([this](https://pythonspeed.com/articles/alpine-docker-python/) which led to [this](https://superuser.com/questions/1219609/why-is-the-alpine-docker-image-over-50-slower-than-the-ubuntu-image) which led to [this](https://lists.alpinelinux.org/~alpine/users/%3C6df8863e77b970b466dbfc9a3a5c2bcec3199f48.camel%40aquilenet.fr%3E)). The metric interval made a bigger difference but it seemed worth it to keep both changes.

## How to verify that this has the expected result

see smoke test time reduced (and thus, CI time). Last run on main took [4m28s](https://app.circleci.com/pipelines/github/honeycombio/honeycomb-opentelemetry-python/322/workflows/3a1374d7-7943-4d25-8156-13d320816892), last run on this branch took [2m13s](https://app.circleci.com/pipelines/github/honeycombio/honeycomb-opentelemetry-python/327/workflows/5740a185-872d-4001-a13e-81e9aab7cc8d)
